python/golem_task_api/testutils/inlinetaskapiservice.py

```diff
@@ -22,43 +22,53 @@ def __init__(
             provider_handler: Optional[ProviderAppHandler] = None,
             provider_lifecycle_handler: Optional[AppLifecycleHandler] = None,
     ) -> None:
-        # get_child_watcher enables event loops in child threads
-        asyncio.get_child_watcher()
         self._work_dir = work_dir
         self._requestor_handler = requestor_handler
         self._requestor_lifecycle_handler = requestor_lifecycle_handler
         self._provider_handler = provider_handler
         self._provider_lifecycle_handler = provider_lifecycle_handler
-        self._thread = None
 
-    def _spawn(self, command: str):
+        self._server_process: Optional[multiprocessing.Process] = None
+
+    def _spawn(self, command: str) -> None:
         loop = asyncio.new_event_loop()
+        loop.add_signal_handler(signal.SIGTERM, loop.stop)
```

Comment:
I suggest doing the same for `signal.SIGHUP` and `signal.SIGINT`

Reply:
We're using `.terminate()` to stop the process which sends `SIGTERM`. Why would the other handlers be needed?

Reply:
It's not like we need them, they can be simply sent to the process. Not insisting.